File: Dawning.Gateway/src/Dawning.Identity.Domain/Aggregates/MultiTenancy/Tenant.cs
using System;

namespace Dawning.Identity.Domain.Aggregates.MultiTenancy
{
    /// <summary>
    /// 租户聚合根
    /// </summary>
    public class Tenant
    {
        /// <summary>
        /// 租户ID
        /// </summary>
        public Guid Id { get; set; }

        /// <summary>
        /// 租户代码（唯一标识，用于URL、Header等）
        /// </summary>
        public string Code { get; set; } = string.Empty;

        /// <summary>
        /// 租户名称
        /// </summary>
        public string Name { get; set; } = string.Empty;

        /// <summary>
        /// 租户描述
        /// </summary>
        public string? Description { get; set; }

        /// <summary>
        /// 绑定域名（可选，用于子域名识别租户）
        /// </summary>
        public string? Domain { get; set; }

        /// <summary>
        /// 联系邮箱
        /// </summary>
        public string? Email { get; set; }

        /// <summary>
        /// 联系电话
        /// </summary>
        public string? Phone { get; set; }

        /// <summary>
        /// 租户Logo URL
        /// </summary>
        public string? LogoUrl { get; set; }

        /// <summary>
        /// 租户配置（JSON格式，存储自定义配置）
        /// </summary>
        public string? Settings { get; set; }

        /// <summary>
        /// 数据库连接字符串（可选，用于独立数据库隔离）
        /// </summary>
        public string? ConnectionString { get; set; }

        /// <summary>
        /// 是否启用
        /// </summary>
        public bool IsActive { get; set; } = true;

        /// <summary>
        /// 订阅计划（如 free, basic, pro, enterprise）
        /// </summary>
        public string Plan { get; set; } = "free";

        /// <summary>
        /// 订阅到期时间
        /// </summary>
        public DateTime? SubscriptionExpiresAt { get; set; }

        /// <summary>
        /// 最大用户数限制
        /// </summary>
        public int? MaxUsers { get; set; }

        /// <summary>
        /// 最大存储空间（MB）
        /// </summary>
        public int? MaxStorageMB { get; set; }

        /// <summary>
        /// 创建时间
        /// </summary>
        public DateTime CreatedAt { get; set; }

        /// <summary>
        /// 更新时间
        /// </summary>
        public DateTime? UpdatedAt { get; set; }

        /// <summary>
        /// 创建者ID
        /// </summary>
        public Guid? CreatedBy { get; set; }

        /// <summary>
        /// 更新者ID
        /// </summary>
        public Guid? UpdatedBy { get; set; }
    }
}

File: apps/admin/src/api/tenant.ts
import axios from '@/api/interceptor';
import type { HttpResponse } from './interceptor';

// Tenant type definitions
export interface Tenant {
  id: string;
  code: string;
  name: string;
  description?: string;
  domain?: string;
  email?: string;
  phone?: string;
  logoUrl?: string;
  settings?: string;
  connectionString?: string;
  isActive: boolean;
  plan: string;
  subscriptionExpiresAt?: string;
  maxUsers?: number;
  maxStorageMB?: number;
  createdAt: string;
  updatedAt?: string;
  createdBy?: string;
  updatedBy?: string;
}

export interface TenantListResponse {
  list: Tenant[];
  pagination: {
    current: number;
    pageSize: number;
    total: number;
  };
}

export interface CreateTenantRequest {
  code: string;
  name: string;
  description?: string;
  domain?: string;
  email?: string;
  phone?: string;
  logoUrl?: string;
  plan?: string;
  maxUsers?: number;
  maxStorageMB?: number;
  subscriptionExpiresAt?: string;
  isActive?: boolean;
}

export interface UpdateTenantRequest {
  code?: string;
  name?: string;
  description?: string;
  domain?: string;
  email?: string;
  phone?: string;
  logoUrl?: string;
  plan?: string;
  maxUsers?: number;
  maxStorageMB?: number;
  subscriptionExpiresAt?: string;
  settings?: string;
  isActive?: boolean;
}

export interface CurrentTenant {
  tenantId?: string;
  tenantName?: string;
  isHost: boolean;
}

// Get current tenant information
export function getCurrentTenant() {
  return axios.get<HttpResponse<CurrentTenant>>('/api/tenant/current');
}

// Get tenant list (paginated)
// Note: Interceptor will unwrap the response, actual return is TenantListResponse
export function getTenantList(params: {
  keyword?: string;
  isActive?: boolean;
  page?: number;
  pageSize?: number;
}) {
  return axios.get<TenantListResponse>('/api/tenant', { params });
}

// Get all tenants
export function getAllTenants() {
  return axios.get<HttpResponse<Tenant[]>>('/api/tenant/all');
}

// Get all active tenants
export function getActiveTenants() {
  return axios.get<HttpResponse<Tenant[]>>('/api/tenant/active');
}

// Get tenant by ID
export function getTenantById(id: string) {
  return axios.get<HttpResponse<Tenant>>(`/api/tenant/${id}`);
}

// Create tenant
export function createTenant(data: CreateTenantRequest) {
  return axios.post<HttpResponse<Tenant>>('/api/tenant', data);
}

// Update tenant
export function updateTenant(id: string, data: UpdateTenantRequest) {
  return axios.put<HttpResponse<Tenant>>(`/api/tenant/${id}`, data);
}

// Delete tenant
export function deleteTenant(id: string) {
  return axios.delete<HttpResponse<void>>(`/api/tenant/${id}`);
}

// Set tenant active status
export function setTenantActive(id: string, isActive: boolean) {
  return axios.put<HttpResponse<void>>(`/api/tenant/${id}/active`, { isActive });
}

// Check if tenant code is available
export function checkTenantCode(code: string, excludeId?: string) {
  return axios.get<HttpResponse<{ available: boolean }>>('/api/tenant/check-code', {
    params: { code, excludeId },
  });
}

// Check if domain is available
export function checkTenantDomain(domain: string, excludeId?: string) {
  return axios.get<HttpResponse<{ available: boolean }>>('/api/tenant/check-domain', {
    params: { domain, excludeId },
  });
}
